app/src/main/java/com/github/se/eventradar/model/repository/message/FirebaseMessageRepository.kt

```diff
@@ -96,9 +98,25 @@ class FirebaseMessageRepository(db: FirebaseFirestore = Firebase.firestore) : IM
       messageHistory: MessageHistory
```

Comment:
Suggested Update:

In the current implementation of FirebaseMessageRepository.kt, the method addMessage() requires an input of an empty MessageHistory when two users have not previously exchanged messages. This method then does not update as expected because it creates a new MessageHistory. To improve this, it would be more functional to modify the method to take in 2 uids, then start by calling getMessages(user1, user2) at the beginning of the interaction. This change would ensure that if a message history between the two users doesn't exist, it is created appropriately, and if it does exist, it is retrieved and updated correctly.

Reply:
Agree, I was afraid to change the function signature since Jalen created it and it had dependencies on tests etc. But this is the ideal structure.

Reply:
The easiest solution is to have `addMessage` behave as it did before (precondition is that a message history is already created and in the repo), and then only check if messages are in message history on the messages overview screen (if there aren't any messages then don't show the message history). This will be implemented in #281.